source/FreeRTOS_DNS_Parser.c

```diff
@@ -373,124 +420,189 @@
-                                        pcRequestedName = ( char * ) &( pucNewBuffer[ xOffset2 ] );
-                                        pxDNSMessageHeader = ( ( DNSMessage_t * ) pucNewBuffer );
+                        /* If this is not a reply to our DNS request, it might an LLMNR
+                         * request. */
+                        if( xApplicationDNSQueryHook( &xEndPoint, xSet.pcName ) )
+                        {
+                            != pdFALSE )
+                            {
+                                int16_t usLength;
+                                NetworkBufferDescriptor_t * pxNewBuffer = NULL;
+                                LLMNRAnswer_t * pxAnswer;
+                                uint8_t * pucNewBuffer = NULL;
+
+                                if( pxNetworkBuffer != NULL )
+                                {
+                                    if( xBufferAllocFixedSize == pdFALSE )
+                                    {
+                                        size_t uxDataLength = uxBufferLength +
+                                                              sizeof( UDPHeader_t ) +
+                                                              sizeof( EthernetHeader_t ) +
+                                                              uxIPHeaderSizePacket( pxNetworkBuffer );
+
+                                        #if ( ipconfigUSE_IPv6 != 0 )
+                                            if( xSet.usType == dnsTYPE_AAAA_HOST )
+                                            {
+                                                uxExtraLength = sizeof( LLMNRAnswer_t ) + ipSIZE_OF_IPv6_ADDRESS - sizeof( pxAnswer->ulIPAddress );
+                                            }
+                                            else
+                                        #endif /* ( ipconfigUSE_IPv6 != 0 ) */
+                                        {
+                                            uxExtraLength = sizeof( LLMNRAnswer_t );
+                                        }
+
+                                        /* Set the size of the outgoing packet. */
+                                        pxNetworkBuffer->xDataLength = uxDataLength;
+                                        pxNewBuffer = pxDuplicateNetworkBufferWithDescriptor( pxNetworkBuffer,
+                                                                                              uxDataLength +
+                                                                                              uxExtraLength );
+
+                                        if( pxNewBuffer != NULL )
+                                        {
+                                            BaseType_t xOffset1, xOffset2;
+
+                                            xOffset1 = ( BaseType_t ) ( xSet.pucByte - pucUDPPayloadBuffer );
+                                            xOffset2 = ( BaseType_t ) ( ( ( uint8_t * ) pcRequestedName ) - pucUDPPayloadBuffer );
```

Comment:
This piece of code come from main. `pcRequestedName` is undefined, shouldn't it be `xSet.pcRequestedName`?

Reply:
Updated.